### ISSUE TITLE
Unable to add rule for ICMP type 0 (Echo Reply) using aws_network_acl_rule

### ISSUE THREAD

**Issue by kwilczynski** [hashicorp/terraform#7269]
At the moment, using `GetOk()` over `schema.TypeInt` with the value of 0 for `icmp_type` (the `icmp_code` is also affected by this) results in the value of 0 being discarded as empty and/or "zero value", as per the following: [resource_aws_network_acl_rule.go#L106-L115](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_network_acl_rule.go#L106-L115)

```go
// Specify additional required fields for ICMP
if p == 1 {
	params.IcmpTypeCode = &ec2.IcmpTypeCode{}
	if v, ok := d.GetOk("icmp_code"); ok {
		params.IcmpTypeCode.Code = aws.Int64(int64(v.(int)))
	}
	if v, ok := d.GetOk("icmp_type"); ok {
		params.IcmpTypeCode.Type = aws.Int64(int64(v.(int)))
	}
}
```

This affects version 0.6.16 and newer, probably since the code was first introduced.

An example:

```hcl
resource "aws_network_acl_rule" "rule_ingress_allow_icmp_echo_reply" {
  network_acl_id = "${module.acl.id}"

  rule_number = 1
  rule_action = "allow"

  protocol   = "icmp"
  cidr_block = "0.0.0.0/0"

  icmp_type = 0
  icmp_code = -1
}
```

Would produce an error:

```
* aws_network_acl_rule.rule_ingress_allow_icmp_echo_reply: Error Creating Network Acl Rule: MissingParameter: The request must contain the parameter icmpTypeCode.type
	status code: 400, request id: 16a1bb97-ced4-4ef7-bc3e-24eca1294f9d
```

I think, that since there is no concept of an "_undefined variable_" that would uniformly be some sort of a special magic value which could be used to indicate that the attribute is indeed undefined, plus changing the way how Schema is designed (especially with the helper functions like the aforementioned `GetOk()`) would be too much of a change, the simplest fix, which I can think of, would be to make these two attributes to be of a string type (as it has its default "zero value" set to be an empty string internally), and then check whether string is not empty. If not, then try `strconv.ParseInt()` over the value and handle error correctly should there be any.

In other words, an empty string would indicate unset value, and any other value that does not cause parsing error would be a valid integer ICMP type, including 0.

**Edit**: List of ICMP types and codes, for posterity: http://www.nthelp.com/icmp.html

**Comment by kwilczynski** [comment]
I just want to add, that @phinze and I had a chat about this already. It should be trivial to fix. I will (when I have some spare time) try to send a PR to address this.

**Comment by phinze** [comment]
Fixing PR is merged! 🎉
